core/eolearn/tests/test_core_tasks.py

```diff
@@ -199,46 +199,38 @@ def test_remove_fails(patch: EOPatch) -> None:
         RemoveFeatureTask((FeatureType.BBOX, None))(patch)
 
 
-def test_duplicate_feature(patch):
-    mask_data = np.arange(10).reshape(5, 2, 1, 1)
-    feature_name = "MASK1"
-    duplicate_name = "MASK2"
-
-    patch = AddFeatureTask((FeatureType.MASK, feature_name))(patch, mask_data)
-
-    duplicate_task = DuplicateFeatureTask((FeatureType.MASK, feature_name, duplicate_name))
-    patch = duplicate_task(patch)
-
-    assert duplicate_name in patch.mask, "Feature was not duplicated. Name not found."
-    assert id(patch.mask["MASK1"]) == id(patch.mask["MASK2"])
-    assert np.array_equal(
-        patch.mask[duplicate_name], mask_data
-    ), "Feature was not duplicated correctly. Data does not match."
+@pytest.mark.parametrize(
+    "feature_specification",
+    [
+        [(FeatureType.DATA, "bands", "bands2")],
+        [(FeatureType.DATA, "bands", "bands2"), (FeatureType.MASK_TIMELESS, "mask", "mask2")],
+        [(FeatureType.DATA, "bands", f"bands{i}") for i in range(5)],
+    ],
+)
+def test_duplicate_feature(feature_specification: List[FeaturesSpecification], patch: EOPatch) -> None:
+    patch = DuplicateFeatureTask(feature_specification)(patch)
 
-    with pytest.raises(ValueError):
-        # Expected a ValueError when creating an already exising feature.
-        patch = duplicate_task(patch)
+    for f_type, f_name, f_dup_name in feature_specification:
+        assert f_dup_name in patch[f_type]
+        assert id(patch[(f_type, f_name)]) == id(patch[(f_type, f_dup_name)]), "DuplicateFeatureTask used deep copy."
+        assert_array_equal(patch[(f_type, f_name)], patch[(f_type, f_dup_name)])
 
-    duplicate_names = {"D1", "D2"}
-    feature_list = [(FeatureType.MASK, "MASK1", "D1"), (FeatureType.MASK, "MASK2", "D2")]
-    patch = DuplicateFeatureTask(feature_list).execute(patch)
 
-    assert duplicate_names.issubset(patch.mask), "Duplicating multiple features failed."
+def test_duplicate_feature_deep(patch: EOPatch) -> None:
+    patch = DuplicateFeatureTask((FeatureType.DATA, "bands", "bands_dup"), deep_copy=True)(patch)
 
-    patch = DuplicateFeatureTask((FeatureType.MASK, "MASK1", "DEEP"), deep_copy=True)(patch)
-    assert id(patch.mask["MASK1"]) != id(patch.mask["DEEP"])
-    assert np.array_equal(
-        patch.mask["MASK1"], patch.mask["DEEP"]
-    ), "Feature was not duplicated correctly. Data does not match."
+    assert "bands_dup" in patch.data
+    assert_array_equal(patch.data["bands_dup"], patch.data["bands"])
+    assert id(patch.data["bands_dup"]) != id(patch.data["bands"])
```

Comment:
either this line is not needed, or the following two (where you change one and check that the other is changed) are not needed